v2/pkg/health/health.go

```diff
@@ -0,0 +1,125 @@
+package health
+
+import (
+	"context"
+	"errors"
+	mongoDriver "github.com/ONSdigital/dp-mongodb/v2/pkg/mongo-driver"
+
+	"github.com/ONSdigital/dp-healthcheck/healthcheck"
+	"github.com/ONSdigital/log.go/log"
+)
+
+// ServiceName mongodb
+const ServiceName = "mongodb"
+
+const timeOutInSeconds = 5
+
+var (
+	// HealthyMessage is the message that will be used in healthcheck when mongo is Healthy and all the collections exist
+	HealthyMessage = "mongodb is OK and all expected collections exist"
+)
+
+//List of errors
+var (
+	ErrorCollectionDoesNotExist = errors.New("collection not found in database")
+	ErrorWithMongoDBConnection  = errors.New("unable to connect with MongoDB")
+)
+
+// Healthcheck health check function
+type Healthcheck = func(context.Context) (string, error)
+
+// CheckMongoClient is an implementation of the mongo client with a healthcheck
+type CheckMongoClient struct {
+	Client      Client
+	Healthcheck Healthcheck
+}
+
+type (
+	//Database a list of mongo types
+	Database string
+	//Collection a list of mongo types
+	Collection string
+)
+
+// Client provides a healthcheck.Client implementation for health checking the service
+type Client struct {
+	mongoConnection    *mongoDriver.MongoConnection
+	serviceName        string
+	databaseCollection map[Database][]Collection
+}
+
+// NewClient returns a new health check client using the given service
+func NewClient(mongoConnection *mongoDriver.MongoConnection) *Client {
+	return NewClientWithCollections(mongoConnection, nil)
+}
+
+// NewClientWithCollections returns a new health check client containing the collections using the given service
+func NewClientWithCollections(mongoConnection *mongoDriver.MongoConnection, clientDatabaseCollection map[Database][]Collection) *Client {
+	return &Client{
+		mongoConnection:    mongoConnection,
+		serviceName:        ServiceName,
+		databaseCollection: clientDatabaseCollection,
+	}
+}
+
+func checkCollections(ctx context.Context, mongoConnection *mongoDriver.MongoConnection, databaseCollectionMap map[Database][]Collection) (err error) {
+
+	for databaseToCheck, collectionsToCheck := range databaseCollectionMap {
+
+		logData := log.Data{"Database": string(databaseToCheck)}
+		collectionsInDb, err := mongoConnection.ListCollectionsFor(ctx, string(databaseToCheck))
+		if err != nil {
+			log.Event(ctx, "Failed to connect to mongoDB to get the collections", log.ERROR, logData, log.Error(err))
```

Comment:
Rather than using log.Event and passing severity level, why not use the wrapper functions log.Error(), log.Info() etc?